src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import React, { useCallback } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setQuery } from '../../features/querySlice';
+import { SearchIconColor } from '../../types/SearchIconColor';
+import { debounce } from '../../utils/debounce';
+import Icon from '../Icon/Icon';
+import styles from './SearchBar.module.scss';
+
+type Props = {
+  setPage?: (value: number) => void;
+  setQueryFromUrl?: (value: string) => void;
+  searchIconColor: SearchIconColor;
+};
+
+export const SearchBar: React.FC<Props> = ({
+  setPage,
+  setQueryFromUrl,
+  searchIconColor,
+}) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { query } = useAppSelector((state) => state.query);
+  const setQueryToApply = (value: string) => {
+    if (setQueryFromUrl && setPage) {
+      setPage(1);
+      setQueryFromUrl(value);
+      dispatch(setQuery(value));
+      searchParams.set('page', String(1));
+      searchParams.set('search', String(value));
+      setSearchParams(searchParams);
+    }
+  };
+  const applyQuery = useCallback(debounce(setQueryToApply, 1000), []);
+  const onHandleQueryInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setQuery(event.target.value));
+
+    if (searchIconColor === 'gray') {
+      applyQuery(event.target.value);
+    }
+  };
+
+  const onHandleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (searchIconColor === 'white') {
```

Comment:
very strange condition. do not use parameters like this searchIconColor to detect if you need to handle submit or not